lib/boundkeypair/experiment/experiment.go

```diff
@@ -0,0 +1,40 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package experiment
+
+import (
+	"os"
+	"sync"
+)
+
+var mu sync.Mutex
+
+var experimentEnabled = os.Getenv("TELEPORT_BOUND_KEYPAIR_JOINING_EXPERIMENT") == "1"
```

Comment:
Can we use `strconv.ParseBool` here just to be consistent with other similar env vars?

You might also consider a `TELEPORT_UNSTABLE_` prefix for the var name to be consistent with similar approaches we've taken in the past. This communicates to the person who enables the feature that it's early and subject to change in the future.

Reply:
I've changed the impl to use `strconv.ParseBool()`. We've been copying and pasting this for our feature flags for a while and I see why we went with the simpler implementation for these hacky flags, handling the error case does make things a bit uglier.

Reply:
I typically don't handle the error case - if you get an error from `ParseBool` you are guaranteed to get `false` (not enabled).

Reply:
Ah, good call, that's much simpler. Thanks!